src/jrd/inf.cpp

```diff
@@ -226,8 +226,9 @@ void INF_database_info(thread_db* tdbb,
  **************************************/
  	CHECK_INPUT("INF_database_info");
 
+	const CountsBuffer::size_type bufferLength = BUFFER_SMALL;
 	CountsBuffer counts_buffer;
-	UCHAR* buffer = counts_buffer.getBuffer(BUFFER_SMALL);
+	UCHAR* buffer = counts_buffer.getBuffer(bufferLength, false);
```

Comment:
One thing worry me in this line: currently getBuffer always return inline storage because CountsBuffer is declared with initial capacity BUFFER_SMALL but nothing ensure that. If getBuffer ever return a pointer to dynamic storage and it is reallocated in process, value of buffer will become invalid.

Reply:
After every counts_buffer's resize variable buffer should be reassigned, see handling of isc_info_read_seq_count and others below.

Reply:
I don't see it in isc_info_db_id handler.

Reply:
Right, it deserves fixing too.

Reply:
Here or another day?

Reply:
Better now, if you don't mind.

Reply:
Ok.